internal/commands/image/formatter.go

```diff
@@ -0,0 +1,126 @@
+package image
+
+import (
+	"time"
+
+	"github.com/docker/cli/cli/command/formatter"
+	"github.com/docker/docker/pkg/stringid"
+	"github.com/docker/go-units"
+)
+
+const (
+	defaultImageTableFormat           = "table {{.Repository}}\t{{.Tag}}\t{{.ID}}\t{{.Name}}\t{{if .CreatedSince }}{{.CreatedSince}}{{else}}N/A{{end}}\t"
+	defaultImageTableFormatWithDigest = "table {{.Repository}}\t{{.Tag}}\t{{.Digest}}\t{{.ID}}\t{{.Name}}\t\t{{if .CreatedSince }}{{.CreatedSince}}{{else}}N/A{{end}}\t"
+
+	imageIDHeader    = "APP IMAGE ID"
+	repositoryHeader = "REPOSITORY"
+	tagHeader        = "TAG"
+	digestHeader     = "DIGEST"
+	imageNameHeader  = "APP NAME"
+)
+
+// NewImageFormat returns a format for rendering an ImageContext
+func NewImageFormat(source string, quiet bool, digest bool) formatter.Format {
+	switch source {
+	case formatter.TableFormatKey:
+		switch {
+		case quiet:
+			return formatter.DefaultQuietFormat
+		case digest:
+			return defaultImageTableFormatWithDigest
+		default:
+			return defaultImageTableFormat
+		}
+	}
```

Comment:
I should have written it like:

```go
if source == formatter.TableFormatKey {
        if quiet {
                return formatter.DefaultQuietFormat
        }
        if digest {
                return defaultImageTableFormatWithDigest
        }
        return defaultImageTableFormat
}
```

I don't see the usage of `switch` with only one case and `switch` on different variables.

Reply:
Because this code was a (simplified) version of the original formatter from docker/cli. Wanted to keep implementations in sync

Reply:
OK, I understand.